### ISSUE TITLE
Full fp16 precision replace Mixed Precision  and use Netron to open .pth file

### ISSUE THREAD

**Issue by mlbo** [NVIDIA-AI-IOT/torch2trt#475]
## 1、Full fp16
I use such code：
`model_trt = torch2trt(model, [data], fp16_mode=True)`
I saw this note
```
When ``fp16_mode=True``, this does not necessarily mean that TensorRT will select FP16 layers.
The optimizer attempts to automatically select tactics which result in the best performance.
```
I don't know **which layer is fp16 and which laye is fp32**
**If I only use fp16 ,What should I do**
And When I use such code：
`model_trt = torch2trt(model, [data], fp16_mode=True  int8_mode=True)`
I guess my model use both fp32, fp16 and int8。 Don't know if it's right

## 2、Using Netron to open saved model
And when I save my model as .pth file
`torch.save(model_trt.state_dict(), 'resnet18_trt_int8.pth')`
 I can't open it by Netron
![image](https://user-images.githubusercontent.com/29347606/103070795-3b5e2480-45fd-11eb-8476-67d09ada061b.png)

My Environment
* Jetpack 4.4.1 [L4T 32.4.4]
* CUDA: 10.2.89
* cuDNN: 8.0.0.180
* TensorRT: 7.1.3.0

**Comment by jaybdub** [comment]
Hi mlbo,

Thanks for reaching out!

## Regarding 1 - Full FP16

If you want to enforce the type constraints, you can set the ``strict_type_constraints=True``.  The TensorRT documentation for this flag is located [here](https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/infer/Core/Builder.html).

So for example,

```python
model_trt = torch2trt(model, [data], fp16_mode=True, strict_type_constraints=True)
```

By default, TensorRT will select the type which results in the highest performance.  In some instances, this may actually be the higher precision layer.

## Regarding 2 - Netron

From what I understand of netron (I've not been familiar), it is a tool for visualizing network graphs.  Unfortunately, the saved TensorRT engine does not expose the underlying network graph (after optimizations / layer fusions, etc.).  If you want to just visualize the PyTorch network, you could export it to ONNX.  If you want to visualize the pre-optimized TensorRT network, we have a utility function for visualizing the TensorRT network definition [here](https://github.com/NVIDIA-AI-IOT/torch2trt/blob/master/torch2trt/utils.py#L5).

You would call

```python
model_trt = torch2trt(model, [data], ...)
trt_network_to_dot_graph(model_trt.network)  # only works before saving model, since model_trt.network is no saved.
```

Please let me know if this helps or if you have any questions.

Best,
John

**Comment by mlbo** [comment]
Hi jaybdub,

Thanks for answering!

I do this, but I find whether use fp32 or fp16 mode，my model recognition accuracy is the same(I don't know if this result is correct). 
And when use fp16 mode my model size  is reduced, But not in half.
